master/internal/task/allocation.go

```diff
@@ -224,6 +226,25 @@ func (a *Allocation) Receive(ctx *actor.Context) error {
 			a.logger.Insert(ctx, a.enrichLog(model.TaskLog{Log: err.Error()}))
 			a.Error(ctx, err)
 		}
+	case WatchAllGather, UnwatchAllGather, allGatherTimeout:
+		if a.allGather == nil {
+			switch msg.(type) {
+			case WatchAllGather:
+				a.allGather = NewAllGather()
+				a.allGather.PreStart(ctx)
+			case UnwatchAllGather, allGatherTimeout:
+				// Ignore without active all gather.
+				return nil
+			}
+		}
+
+		switch finished, err := a.allGather.ReceiveMsg(ctx); {
```

Comment:
ah, so there's no explicit `ctx.Respond` here because you call `ReceiveMsg` directly, which does `ctx.Respond` internally. 
so the `AllGather` is not really an actor of its own anyway, even though it looks like one. another reason for it not to be "actor-y".